backend/src/api.py

```diff
@@ -0,0 +1,236 @@
+from __future__ import annotations
+
+import importlib
+import os
+from dataclasses import dataclass, field
+from typing import Any, Callable, Dict, List, Tuple, TypeVar, Union
+
+from sanic.log import logger
+
+from base_types import InputId, OutputId
+from nodes.group import Group, GroupId, NestedGroup, NestedIdGroup
+from nodes.node_base import NodeType
+from nodes.properties.inputs.base_input import BaseInput
+from nodes.properties.outputs.base_output import BaseOutput
+
+
+@dataclass
+class Node:
+    schema_id: str = ""
+    description: str = ""
+    name: str = ""
+    icon: str = ""
```

Comment:
Why do all fields have default values? Most of those don't even make sense. E.g. `""` isn't a valid schema id.